app/services/version_batch_status_service.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+# Service for retrieving the version status of a batch of objects.
+class VersionBatchStatusService
+  # This isn't a WorkflowStateService, but it acts like one.
+  DummyWorkflowStateService = Struct.new('DummyWorkflowStateService', :accessioning, :accessioned, :assembling) do
+    def accessioning?
+      accessioning
+    end
+
+    def accessioned?
+      accessioned
+    end
+
+    def assembling?
+      assembling
+    end
+  end
+
+  def self.call(...)
+    new(...).call
+  end
+
+  # raises CocinaObjectStore::CocinaObjectNotFoundError if the druid is not found
+  def self.call_single(druid:)
+    new(druids: [druid]).call[druid].tap do |status|
```

Comment:
Nothing wrong with this, but it does look kinda funky. I did a double-take anyway. Could consider a tweak like:


```suggestion
    new(druids: [druid]).call.fetch(druid, nil).tap do |status|
```